pkg/netceptor/netceptor.go

```diff
@@ -1165,7 +1165,12 @@ func (s *Netceptor) forwardMessage(md *MessageData) error {
 	// decrement HopsToLive
 	message[1]--
 	logger.Trace("    Forwarding data length %d via %s\n", len(md.Data), nextHop)
-	c.WriteChan <- message
+
+	select {
+	case c.WriteChan <- message:
+	case <-time.After(1 * time.Second):
```

Comment:
@ghjm Mind giving me your thoughts on this? I found it when debugging this flaky test: 

https://github.com/ansible/receptor/blob/3000b6175bc31851017cd6004ed80bb1485ccd49/tests/functional/mesh/work_test.go#L327-L331

Notice how we shut down node2 and then immediately try to submit work.

We get hung up here on this write to `WriteChan` via: https://github.com/ansible/receptor/blob/3000b6175bc31851017cd6004ed80bb1485ccd49/pkg/netceptor/conn.go#L324

I was unable to use a context here like I did in the blob below because the node unexpected goes away. I thought maybe we could use a write deadline, but noticed this:

https://github.com/ansible/receptor/blob/3000b6175bc31851017cd6004ed80bb1485ccd49/pkg/netceptor/packetconn.go#L234

![](https://miro.medium.com/max/910/1*snTXFElFuQLSFDnvZKJ6IA.png)

Reply:
See my latest commit. The context didn't seem to be enough there either.